syne_tune/optimizer/schedulers/searchers/searcher.py

```diff
@@ -612,7 +689,13 @@ def _generate_all_candidates_on_grid(self) -> List[dict]:
             all_candidates_on_grid.append(dict(zip(hp_keys, values)))
```

Comment:
This is not your code, so OK to leave it. But I find it pretty wasteful to maintain this huge number of dicts. It is already not that great to create `hp_values_combinations` explicitly, but at least it should remain a list of tuples.